hstream-kafka/HStream/Kafka/Server/Handler/Basic.hs

```diff
@@ -178,7 +178,7 @@ handleMetadata ctx reqCtx req = do
              , partitions = K.emptyKaArray
              , isInternal = False
              }
-      shards_e <- try ((V.map snd) <$> S.listStreamPartitionsOrdered ctx.scLDClient streamId)
+      shards_e <- try ((V.map snd) <$> S.listStreamPartitionsOrderedByName ctx.scLDClient streamId)
```

Comment:
I don't think it's a good idea to directly call the underlying `listStreamPartitionsOrderedByName` here. Within the Kafka server, the terms used are `topic`, `partition`, `topic name`, and `partitionId`. IMO using the underlying storage's corresponding terms, `stream`, `log name`... is actually a manifestationan of abstraction leak. That is why I extracted a separate store package in my implementation. 

Moreover, the log name is completely irrelevant to the Kafka server layer. The returned results are ordered, and based on this condition, we naturally know what each logId's corresponding partitionId should be. Therefore, I think it would be better to return an vector of logIds rather than a tuple.

Reply:
You can open another PR to do this all. But for this PR, I'll keep the changes as small as possible.

Reply:
Also, `HStream.Store.Stream` is already an abstract layer. All tops can use it. There is no need to add a new module to do this. You can edit `HStream.Store.Stream` directly.

Reply:
The current server-level store package is just a basic implementation; if we want to abstract away the underlying storage with interfaces or something else in the future, then the specific interfaces and implementations can be placed in this package